consensus/core/src/core.rs

```diff
@@ -274,8 +269,24 @@ impl Core {
         // only when the validator was supposed to be the leader of the round - so we bring down the missed leaders.
         // Probably proposing for all the intermediate rounds might not make much sense.
 
-        // Consume the ancestors to be included in proposal
-        let ancestors = self.ancestors_to_propose(clock_round, now);
+        // Determine the ancestors to be included in proposal
+        let ancestors = self.ancestors_to_propose(clock_round);
+        self.context
+            .metrics
+            .node_metrics
+            .block_ancestors_count
+            .inc_by(ancestors.len() as u64);
+
+        // Ensure ancestor timestamps are not more advanced than the current time.
+        // Also catch the issue if system's clock go backwards (ex due to ntp corrections)
```

Comment:
regarding the ntp corrections, I explicitly had that looked up and it looks like ntp will never attempt setting the clock backwards . But I assume things can still happen from some manual time change